src/common.speech/Transcription/ConversationServiceAdapter.ts

```diff
@@ -0,0 +1,627 @@
+import { ConversationTranslatorRecognizer } from "./ConversationTranslatorRecognizer";
+import {
+    CommandResponsePayload,
+    IParticipantPayloadResponse,
+    IParticipantsListPayloadResponse,
+    ITranslationResponsePayload,
+    ParticipantPayloadResponse,
+    ParticipantsListPayloadResponse,
+    SpeechResponsePayload,
+    TextResponsePayload  } from "./ServiceMessages/Exports";
+
+/***
+ * The service adapter handles sending and receiving messages to the Conversation Translator websocket.
+ */
+export class ConversationServiceAdapter extends ServiceRecognizerBase {
+    private privConversationServiceConnector: ConversationTranslatorRecognizer;
+    private privConversationConnectionFactory: IConnectionFactory;
+    private privConversationAuthFetchEventId: string;
+    private privConversationIsDisposed: boolean;
+    private privConversationAuthentication: IAuthentication;
+    private privConversationRequestSession: ConversationRequestSession;
+    private privConnectionConfigPromise: Promise<IConnection>;
+    private privConversationConnectionPromise: Promise<IConnection>;
+    private privConnectionLoop: Promise<IConnection>;
+    private terminateMessageLoop: boolean;
+    private privUtteranceId: string = "";
+
+    public constructor(
+        authentication: IAuthentication,
+        connectionFactory: IConnectionFactory,
+        audioSource: IAudioSource,
+        recognizerConfig: RecognizerConfig,
+        conversationServiceConnector: ConversationTranslatorRecognizer) {
+
+        super(authentication, connectionFactory, audioSource, recognizerConfig, conversationServiceConnector);
+
+        this.privConversationServiceConnector = conversationServiceConnector;
+        this.privConversationAuthentication = authentication;
+        this.receiveMessageOverride = this.receiveConversationMessageOverride;
+        this.recognizeOverride = this.noOp;
+        this.connectImplOverride = this.conversationConnectImpl;
+        this.configConnectionOverride = this.configConnection;
+        this.fetchConnectionOverride = this.fetchConversationConnection;
+        this.disconnectOverride = this.privDisconnect;
+        this.privConversationRequestSession = new ConversationRequestSession(createNoDashGuid());
+        this.privConversationConnectionFactory = connectionFactory;
+        this.privConversationIsDisposed = false;
+    }
+
+    public isDisposed(): boolean {
+        return this.privConversationIsDisposed;
+    }
+
+    public dispose(reason?: string): void {
+        this.privConversationIsDisposed = true;
+        if (this.privConnectionConfigPromise) {
+            this.privConnectionConfigPromise.onSuccessContinueWith((connection: IConnection) => {
+                connection.dispose(reason);
+            });
+        }
+    }
+
+    public sendMessage = (message: string): Promise<boolean> => {
```

Comment:
You're replacing not overrriding the base method and making it return a Promise. However, in its current state, this method always immediately returns a completed promise even though it has work that has not completed yet.

Reply:
This was based off DialogServiceAdapter - have removed the dispose override.

Reply:
I was referring to the sendMessage method rather than the dispose() method